augur/parse.py

```diff
@@ -176,6 +176,8 @@ def run(args):
                 args.prettify_fields,
                 args.fix_dates
             )
+            if sequence_record.id in meta_data:
+                raise AugurError(f"Duplicate found for '{sequence_record.id}'.")
```

Comment:
Another more comprehensive approach - the solution outlined in #616 - is to run through all sequences to capture a list of all duplicates, then error with a complete list of duplicates. However, this would require either:

1. one pass through the data + large memory usage
2. two passes with minimal memory usage (1 for duplicate detection, one for writing outputs if no duplicates)
3. one pass through + temporary files that are moved to the proper location if no duplicates are found

but I'm reluctant to use that as the initial implementation unless a complete list of duplicates is really desired from a user perspective.

I think this small change is sufficient to address the inaccurate behavior described by #616.